storyscript/compiler/semantics/types/Types.py

```diff
@@ -626,10 +675,29 @@ def object(self):
     def op(self, op):
         return None
 
-    def index(self, other, kind):
+    def index(self, tree, symbol, name, other, kind):
         if kind == IndexKind.DOT:
-            return AnyType.instance()
-        if other.implicit_to(StringType.instance()) is not None:
+            assert isinstance(other, StringType)
+            if isinstance(self._object, ServiceOutput):
+                prop = self._object.property(name)
+                tree.expect(prop is not None, 'service_output_invalid_prop',
+                            object=symbol.name(), prop=name)
+                return get_type_instance(var=prop)
+            if isinstance(self._object, dict):
+                # right now we are only storing a dict as the wrapped
+                # value inside a ObjectType instance for the special case
+                # of `app` keyword.
+                assert symbol.name() == 'app'
+                prop = self._object.get(name, None)
+                tree.expect(prop is not None, 'object_invalid_prop',
+                            object=symbol.name(), prop=name)
+                # for the app we already store the correct TypeClass in Symbol
+                # type therefore no need to perform mapping.
+                return prop.type()
+            if isinstance(self._object, MapType):
+                return self._object.index(other, IndexKind.INDEX)
+            # if the object isn't a service output object, do no checks on
+            # property existance for now.
             return AnyType.instance()
```

Comment:
Why not return `None` now here? No tests fail anymore.
```suggestion
            return None
```